test/integration/timeout.integration.test.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * © Copyright IBM Corporation 2021. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+
+const { CloudantV1 } = require('../../index.ts');
```

Comment:
I realise we do this elsewhere, but it seems a little strange to me; is it right to be importing `ts` in a `js` test?

The `cloduant.v1.test` pulls js from the `dist` folder instead. Perhaps that is what we should be doing here too and we can follow up on the other test places after?

Reply:
I think this would be better to leave as-it-is, because the `index.ts` defines the `cloudant-node-sdk` module. This file defines the content of a module, what is publicly available from it and what is private/hidden.
So if you want to test whether something is exported/available in the module, then better to do this type of import.
Since it is an `integration` test, I would stay to use this type of referring. The `dist` folder is created during deployment, so if there is a need to test a new feature with integration test, then after the changes a local deploy should have been executes, this change would just make the development unnecessarily cumbersome.

Reply:
Fair point that it is nominally integration test, but I don't think there is a universal definition of a formal distinction between `unit` and `integration`. I think our `unit` vs `integration` are actually split based on whether they need a server or not. Given these new tests don't reqiure a running server my preference would be that they live with the `unit` tests anyway.

WRT to the import I don't think it is any more cumbersome - we already make `dist` before the tests anyway as part of `npm run build`
e.g. `test-unit`
https://github.com/IBM/cloudant-node-sdk/blob/68d9a2c7da58ae914e0dc26020033e68e9824de7/package.json#L26
and
https://github.com/IBM/cloudant-node-sdk/blob/68d9a2c7da58ae914e0dc26020033e68e9824de7/package.json#L20

but as I observed we already use the `ts` import elsewhere even in other "unit" tests and I can live with this doing the same thing.

So how about we leave the import as is, but move them to the unit tests folder does that work for you?

Reply:
> we already make dist before the tests anyway

I am not part if this `we` group, I test differently. I do not know about any rule that we agreed how the team have to use a testing tool.

> So how about we leave the import as is, but move them to the unit tests folder; does that work for you?

Not really, because this tests checks not just a class/unit/component, it creates a full `CloudantV1` and checks the settings impact on the authenticators. So it has effects on more than 1 unit.

Reply:
I can introduce a `module` testing section and I can move it there, since it checks the behavior of the whole module, maybe.

Reply:
>this tests checks not just a class/unit/component, it creates a full CloudantV1

so does `unit/cloudant.v1.test.js`

I'd rather it just sayed here in `integration` than introducing yet another test bucket and series of commands to run.
FWIW 5/6 test files in our `unit` folder import something from one of the dependencies so fail in terms of "unit" isolation in the truest sense, but I'm disinclined to get hung up on this point.